src/map.ts

```diff
@@ -113,6 +113,55 @@ export async function mapLimit<V>(input: any, limit: number, iteratee: any): Pro
   return results;
 }
 
+/**
+ * The same as map limit but it allows the iteratee function to recieve additional arguments.
+ *
+ * @param {Array | Iterable} input - A collection to iterate over.
+ * @param {number} limit - The maximum number of async operations at a time.
+ * @param {AsyncFunction} iteratee - An async function to apply to each item in coll. The iteratee
+ *     should complete with the transformed item. Invoked with (item, key).
+ * @param {any} args - Various arguments that the iteratee will use to function.
+ */
+export async function mapLimitWithArgs<T, V>(
+  input: readonly T[],
+  limit: number,
+  // tslint:disable-next-line: no-any
+  iteratee: (value: T, ...args: any) => Promise<V>,
+  // tslint:disable-next-line: no-any trailing-comma
+  ...args: any
+): Promise<V[]> {
+  if (!input) {
+    return [];
+  }
+
+  const size = input.length;
+  const allValues = new Array(size);
+  const results = new Array(size);
+
+  let i = 0;
+  for (const key in input) {
+    const possiblyNumericKey = i;
+    allValues[size - 1 - i] = [input[key], i, possiblyNumericKey];
+    ++i;
+  }
+
+  const execute = async () => {
+    while (allValues.length > 0) {
+      // tslint:disable-next-line:no-any
+      const [val, index] = allValues.pop();
+      results[index] = await iteratee(val, ...args);
```

Comment:
I'm not understanding why i would want to use this function. Because of how this is generally invoked, why would I not just be able to refer to these args directly? E.g. in the test below, `argNum` is a variable in scope, so instead of calling `mapLimitWithArgs`, I could just call `promiseUtils.mapLimit(_.range(maxRangeNum), 5, async n => n*n*argNum)`

Reply:
Re-reviewing this, I was thinking that I was thinking of how I was passing the function like:
```typescript
await mapLimitWithArgs(
    array,
    limit,
    promiseFunction,
    arg1,
    arg2,
    arg3,
  );
```

instead of:

```typescript
await mapLimit(
    array,
    limit,
    async (item) => promiseFunction(item, arg1, arg2, arg3)
  );
```

as you are saying. So after this I think I'll drop this PR since its no longer needed.

Thanks for the feedback.